File: LeetCode/LeetCode/contains-duplicate.cc
#include <unordered_set>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
    public:
        bool containsDuplicate(vector<int>& nums) {
            unordered_set<int> nums_set(nums.begin(), nums.end());
            return nums_set.size() != nums.size();
        }
};


class Solution2 {
    public:
        bool containsDuplicate(vector<int> &nums) {
            sort(nums.begin(), nums.end());
            return unique(nums.begin(), nums.end()) != nums.end();
        }
};

#include<iostream>

int main()
{
    Solution s;
    Solution2 s2;
    vector<int> v{1,2,3,1};
    cout << s.containsDuplicate(v) <<endl;
    cout << s2.containsDuplicate(v) <<endl;
}

File: LeetCode/LeetCode/contains-duplicate.go
package main

import "fmt"

func containsDuplicate(nums []int) bool {
	m := make(map[int]bool, len(nums))
	for _, n := range nums {
		if m[n] {
			return true
		}

		m[n] = true
	}

	return false
}

func main() {
	v := []int{1,2,3,4,1,2}
	v1 := []int{1,2,3,4}
	fmt.Println(containsDuplicate(v))
	fmt.Println(containsDuplicate(v1))
}
